powerdnsadmin/routes/index.py

```diff
@@ -592,26 +592,34 @@ def prepare_welcome_user(user_id):
 
 @index_bp.route('/logout')
 def logout():
-    if current_app.config.get(
-            'SAML_ENABLED'
-    ) and 'samlSessionIndex' in session and current_app.config.get(
-            'SAML_LOGOUT'):
+    if Setting().get('saml_enabled'
+        ) and 'samlSessionIndex' in session and Setting().get(
+            'saml_logout'):
         req = saml.prepare_flask_request(request)
         auth = saml.init_saml_auth(req)
-        if current_app.config.get('SAML_LOGOUT_URL'):
+        if Setting().get('saml_logout_url'):
+            try:
+                return redirect(
+                    auth.logout(
+                        name_id_format=
+                        Setting().get('saml_nameid_format'),
+                        return_to=Setting().get('saml_logout_url'),
+                        session_index=session['samlSessionIndex'],
+                        name_id=session['samlNameId']))
+            except:
+                current_app.logger.info(
+                "SAML: Your IDP does not support Single Logout.")
+        try:
             return redirect(
                 auth.logout(
                     name_id_format=
                     "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
                     return_to=current_app.config.get('SAML_LOGOUT_URL'),
                     session_index=session['samlSessionIndex'],
                     name_id=session['samlNameId']))
```

Comment:
I don't understand why this stays. You check if there is a logout url in the settings above. If not, it means it should not be tried, doesn't it ?

I also note that you're calling current_app.config.get('SAML_LOGOUT_URL'), it should be fetched from settings as well.

Reply:
Hello,

this old block of code slipped through us while rewriting the PR, that's why the current_app.config.get() statement is still there.

Thank you for pointing this out!